extensions/amp-live-list/0.1/amp-live-list.js

```diff
@@ -0,0 +1,363 @@
+    this.manager_ = installLiveListManager(this.win);
+
+    /** @private @const {string} */
+    this.liveListId_ = user.assert(this.element.getAttribute('id'),
+        'amp-live-list must have an id.');
+
+    /** @private @const {number} */
+    this.pollInterval_ = getNumberOrDefault(
+        this.element.getAttribute('data-poll-interval'),
+        LiveListManager.getMinDataPollInterval());
+
+    const maxItems = this.element.getAttribute('data-max-items-per-page');
+    user.assert(Number(maxItems) > 0,
+        `amp-live-list#${this.liveListId_} must have ` +
+        `data-max-items-per-page attribute with numeric value. ` +
+        `Found ${maxItems}`);
+
+    /** @private @const {number} */
+    this.maxItemsPerPage_ = getNumberOrDefault(maxItems,
+        LiveListManager.getMinDataMaxItemsPerPage());
+
+    /** @private @const {!Object<string, string>} */
+    this.knownChildIds_ = Object.create(null);
+
+    this.manager_.register(this.liveListId_, this);
+
+    this.insertFragment_ = this.win.document.createDocumentFragment();
+
+    this.updateSlot_ = user.assert(
+       this.getUpdateSlot_(this.element),
+       'amp-live-list must have an `update` slot.');
+
+    this.itemsSlot_ = user.assert(
+        this.getItemsSlot_(this.element),
+        'amp-live-list must have an `items` slot.');
+
+    this.updateSlot_.classList.add('-amp-hidden');
+    this.eachChildElement_(this.itemsSlot_, item => {
+      item.classList.add(classes.ITEM);
+    });
+
+    this.validateLiveListItems_(this.itemsSlot_, true);
+
+    this.registerAction('update', this.update_.bind(this));
+  }
+
+  /** @override */
+  update(updatedElement) {
+    const container = this.getItemsSlot_(updatedElement);
+    this.validateLiveListItems_(container);
+    const mutateItems = this.getUpdates_(container);
+
+    // Insert/new items will be contiguous at the top even though they
+    // weren't in the actual request DOM structure.
+    mutateItems.insert.forEach(child => {
+      child.classList.add(classes.ITEM);
+      child.classList.add(classes.NEW_ITEM);
+      this.insertFragment_.insertBefore(child,
+          this.insertFragment_.firstElementChild);
+    });
+
+    if (mutateItems.insert.length > 0) {
+      const isHidden = this.updateSlot_
+          .classList.contains('-amp-hidden');
+      if (isHidden) {
+        this.updateSlot_.classList.remove('-amp-hidden');
+      }
+    }
+  }
+
+  /**
+   * Mutates the current elements dom and compensates for scroll
+   * change if necessary.
+   * @private
+   */
+  update_() {
+    if (this.insertFragment_.childElementCount == 0) {
+      return;
+    }
+
+    // TODO(erwinm): do in place update as well as sorting,
+    // correct insertion, tombstoning etc.
+    this.getVsync().mutate(() => {
+      // Remove the new class from the previously inserted items.
+      this.eachChildElement_(this.itemsSlot_, child => {
+        child.classList.remove(classes.NEW_ITEM);
```

Comment:
so we add it to any newly inserted item on this latest update so that a dev can add highlighting css for example, and when new updates come in we remove this class.